### ISSUE TITLE
Publishing to BAR fails with 

### ISSUE THREAD

**Issue by rynowak** [dotnet/arcade#1452]
I'm following the instructions at https://github.com/dotnet/arcade/blob/master/Documentation/DependencyFlowOnboarding.md#3-enable-assets-publishing-to-the-build-asset-registry-bar

After completing step 3 and triggering an official build I see a whole bunch of failures like:

```
2018-12-05T02:03:38.9838714Z D:\a\1\s\src\Razor\src\RazorPageGenerator\RazorPageGenerator.csproj : error MSB4057: The target "Execute" does not exist in the project.
2018-12-05T02:03:39.0767587Z D:\a\1\s\src\Razor\src\Microsoft.AspNetCore.Razor.Language\Microsoft.AspNetCore.Razor.Language.csproj : error MSB4057: The target "Execute" does not exist in the project.
2018-12-05T02:03:39.2117252Z D:\a\1\s\src\Razor\test\Microsoft.AspNetCore.Razor.Language.Test\Microsoft.AspNetCore.Razor.Language.Test.csproj : error MSB4057: The target "Execute" does not exist in the project.
2018-12-05T02:03:39.2117741Z D:\a\1\s\src\Razor\src\Microsoft.CodeAnalysis.Razor.Workspaces\Microsoft.CodeAnalysis.Razor.Workspaces.csproj : error MSB4057: The target "Execute" does not exist in the project.
```

https://dnceng.visualstudio.com/internal/_build/results?buildId=52608

**Comment by markwilkie** [comment]
cc/ @mmitche @jcagme

**Comment by jcagme** [comment]
@rynowak seems like the arcade version you are using is kinda old. Could you please update the value of `Microsoft.DotNet.Arcade.Sdk` in `global.json` to `1.0.0-beta.18604.15` and try again

**Comment by rynowak** [comment]
This version fails to build my repo. Please advise https://github.com/aspnet/AspNetCore-Tooling/pull/18

```
2018-12-06T18:50:28.9161337Z F:\vsagent\8\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.18604.15\tools\Sign.proj(82,9): error MSB4064: The "DoStrongNameCheck" parameter is not supported by the "Microsoft.DotNet.SignTool.SignToolTask" task. Verify the parameter exists on the task, and it is a settable public instance property.
2018-12-06T18:50:28.9166417Z F:\vsagent\8\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.18604.15\tools\Sign.proj(79,5): error MSB4063: The "Microsoft.DotNet.SignTool.SignToolTask" task could not be initialized with its input parameters. 
2018-12-06T18:50:28.9760118Z 
2018-12-06T18:50:28.9786532Z Build FAILED.
2018-12-06T18:50:28.9787206Z 
2018-12-06T18:50:28.9790257Z F:\vsagent\8\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.18604.15\tools\Sign.proj(82,9): error MSB4064: The "DoStrongNameCheck" parameter is not supported by the "Microsoft.DotNet.SignTool.SignToolTask" task. Verify the parameter exists on the task, and it is a settable public instance property.
2018-12-06T18:50:28.9790984Z F:\vsagent\8\s\.packages\microsoft.dotnet.arcade.sdk\1.0.0-beta.18604.15\tools\Sign.proj(79,5): error MSB4063: The "Microsoft.DotNet.SignTool.SignToolTask" task could not be initialized with its input parameters. 
```

**Comment by JohnTortugo** [comment]
That version of the sdk has this problem indeed. Please, use this one and it will fix the `DoStrongNameCheck` problem:  `Microsoft.DotNet.Arcade.Sdk - 1.0.0-beta.18605.14`

**Comment by JohnTortugo** [comment]
Hi @rynowak , looks like it worked, [right](https://dnceng.visualstudio.com/public/_build/results?buildId=53720)? Are you OK with me closing this issue?

**Comment by rynowak** [comment]
No, that PR was just to update arcade, I have to reenable the build step that pushes to BAR and see that pass in an official build before we will know that this is fixed.

**Comment by rynowak** [comment]
This is failing with the same error message: https://dnceng.visualstudio.com/internal/_build/results?buildId=53950

**Comment by mmitche** [comment]
This is because of this: https://github.com/aspnet/AspNetCore-Tooling/blob/master/eng/Build.props#L4

This adds the solution unconditionally, even when the Execute target is being sent.  The Execute target only applies to Sdk task projects.

@tmat What do you think the right solution here is.  It seems hacky to check the global properties (SdkTaskProjects != '' or Execute==true) before adding the solution to ProjectsToBuild.

**Comment by markwilkie** [comment]
cc/ @chcosta

**Comment by chcosta** [comment]
I'm going to look at refactoring this today / tomorrow (realistically may slip until after the holidays though).  Essentially, the SdkTaskProjects concept was a step in the right direction in terms of isolating those projects from a repos state, but they don't go far enough.

**Comment by safern** [comment]
I'm hitting this as well in CoreFX and had to define an empty Execute target in my Build.proj. The way it reproduces to me is by just calling PublishBuildAssets.cmd

**Comment by mmitche** [comment]
@safern @rynowak Is this still an issue?

**Comment by chcosta** [comment]
I'm waiting until we get through some noisy Arcade official build issues before I merge the fix for this.  Post merge, it's likely to take two rounds of builds to get Arcade correctly work.

**Comment by safern** [comment]
In corefx we still have the Execute target defined in our main .proj, so I haven't tried without it. Do we expect it to be fixed? If so I can try removing the target.

**Comment by rynowak** [comment]
This is fixed for me.

**Comment by chcosta** [comment]
Fixed with Arcade version 1.0.0-beta.19067.5
